pyproject.toml

```diff
@@ -1,3 +1,11 @@
+[build-system]
+    build-backend = "setuptools.build_meta"
+    requires = [
+        "setuptools>=47",
+        "wheel",
+        "cython; python_implementation == 'CPython'", # Skip cython when using pypy
```

Comment:
FWIW:
```shell
$ pyston -c "import platform; print(platform.python_implementation(), platform.python_version())"
CPython 3.8.8
```

Reply:
Well this looks like a problem of pyston.. it should not report that it is cython

Reply:
It's up to them -- if they want to position Pyston as a drop-in replacement for 3.8.8, maybe it's OK.

Reply:
If there is a way of detecting it we can add it, just to be sure.

Reply:
Yes (see also Gitter); ideally we would want to cythonize only `falcon.cyutils`, and not even sure about all functions there. But at least `parse_query_string` is still more efficient than pure Python, even on Pyston.

Reply:
It makes sense adding special handling of pyston, but probably in the future.

Reply:
Meanwhile, I filed an issue with Pyston, if we decide to optimize for it: https://github.com/pyston/pyston/issues/36.